testplan/report/testing/base.py

```diff
@@ -552,6 +552,7 @@ class TestGroupReport(BaseReportGroup):
 
     def __init__(self,
                  name,
+                 display_name=None,
```

Comment:
We definitely don't need an additional display_name parameter on the report. If a custom name is specified for a testcase/testsuite, that becomes the `name` parameter. Only if no name is specified do we use the function name as a default value. The report object only ever has one name.

Reply:
Yes the display name should be the name of testcase here, my concern is , if we totally remove that original function name from report, then it is hard for us to debug or trace with that json report, or maybe one day use want a feature that switch back to testcase identifier on UI. It makes sense that the name of report is 'display name', just like that name of Multitest is a human readable string, but somehow I think we should also keep an attribute 'testcase' or 'testcase_name' in the report.

Reply:
I prefer that we keep the original testcase name (func name) unchanged in the report, just add one additional field say display_name. Changing the name itself might have other side effect that we cannot predict right now.

Reply:
No, as discussed absolutely we cannot have more than one name in the report as it will over complicate the UI code too much. One name only please.

Reply:
I think we have different point of views.
**Ryan**: only a `name` attribute added into report, inside testcase it's actually __display_name__, no redundant attributes kept in report.
**Yifan**: `name` is intuitively the defined name of testcase, so a `display_name` is added in both testcase decorator and test report.
**Me**: `name` is added for testcase decorator because it's short and regular, but in json report we keep both original name and display name, when users search on UI, I think both of them are needed, just like when you search a people, the real name and user id will both be considered.

Reply:
I don't really care too much about what attributes you set on the testcase functions but, to reiterate, we must not overcomplicate the report by adding an extra "display_name" field. The name parameter of testcase is an override, that becomes the testcases unique identifier and is the only name used for search and filter.